libraries/ms-common/src/x86/uart.c

```diff
@@ -1,13 +1,52 @@
 #include "uart.h"
+#include <stdlib.h>
+#include <string.h>
+#include "log.h"
+#include "x86_socket.h"
+
+typedef struct {
+  UARTPort port;
+  X86SocketThread thread;
+  UARTStorage *storage;
+} UartSocketData;
+
+static UartSocketData s_sock[NUM_UART_PORTS];
+
+void uart_receiver_wrapper(struct X86SocketThread *thread, int client_fd, const char *rx_data,
```

Comment:
You don't need to forward-declare the function if you're actually declaring it right afterwards.